src/fidesops/api/v1/endpoints/connection_type_endpoints.py

```diff
@@ -27,41 +29,55 @@
 logger = logging.getLogger(__name__)
 
 
-def get_connection_types(search: Optional[str] = None) -> List[str]:
+def get_connection_types(search: Optional[str] = None) -> List[ConnectionSystemTypeMap]:
     def is_match(elem: str) -> bool:
         """If a search query param was included, is it a substring of an available connector type?"""
         return search in elem if search else True
 
-    database_types: List[str] = [
-        conn_type.value
-        for conn_type in ConnectionType
-        if conn_type
-        not in [ConnectionType.saas, ConnectionType.https, ConnectionType.manual]
-        and is_match(conn_type.value)
-    ]
-    saas_types: List[str] = [
-        saas_type.value
-        for saas_type in SaaSType
-        if saas_type != SaaSType.custom and is_match(saas_type.value)
-    ]
-    connection_types: List[str] = sorted(database_types + saas_types)
+    connection_system_types: List[ConnectionSystemTypeMap] = []
+    database_types: List[str] = sorted(
+        [
+            conn_type.value
+            for conn_type in ConnectionType
+            if conn_type
+            not in [ConnectionType.saas, ConnectionType.https, ConnectionType.manual]
+            and is_match(conn_type.value)
+        ]
+    )
+    connection_system_types.extend(
+        [
+            ConnectionSystemTypeMap(identifier=item, type=SystemType.database)
+            for item in database_types
+        ]
+    )
+    saas_types: List[str] = sorted(
+        [
+            saas_type.value
+            for saas_type in SaaSType
+            if saas_type != SaaSType.custom and is_match(saas_type.value)
+        ]
+    )
+    connection_system_types.extend(
+        [
+            ConnectionSystemTypeMap(identifier=item, type=SystemType.saas)
+            for item in saas_types
+        ]
+    )
 
-    return connection_types
+    return connection_system_types
 
 
 @router.get(
     CONNECTION_TYPES,
     dependencies=[Security(verify_oauth_client, scopes=[CONNECTION_TYPE_READ])],
-    response_model=Page[str],
+    response_model=List[ConnectionSystemTypeMap],
 )
 def get_all_connection_types(
-    *, params: Params = Depends(), search: Optional[str] = None
-) -> AbstractPage[str]:
+    *, search: Optional[str] = None
+) -> List[ConnectionSystemTypeMap]:
     """Returns a list of connection options in Fidesops - includes only database and saas options here."""
 
-    connection_types: List[str] = get_connection_types(search)
-
-    return paginate(connection_types, params)
```

Comment:
Is there a technical constraint that requires us to remove the pagination?

Reply:
@pattisdr had suggested this in the ticket- https://github.com/ethyca/fidesops/issues/910 

Alongside this, the server-side logic to get all connection types for these endpoints isn't costly enough and the data not large enough to really require pagination.

Reply:
I specified no pagination in the endpoint just to support @adriaaaa's acceptance criteria in the frontend for adding a connector

https://github.com/ethyca/fidesops/issues/760

> No pagination, just display an entire list of supported connectors

Reply:
@seanpreston good point with the consistency. I'm OK adding it back, just let me know which you'd prefer.

Reply:
Thanks! I prefer it in, let's keep consistency unless there's a really clear incentive not to. In this case we can have both client side awareness of the whole set and pagination for relatively low cost, so let's keep it

Reply:
@seanpreston @eastandwestwind @pattisdr - one concern about pagination though is how we search for integrations from the front end. We specified that it would be that "typeahead" method in refinement. If we do pagination on this page, would we be limited to the results on that page and therefore it would be considered an incomplete search? May not be the case, but I do just want to confirm.

Reply:
@eastandwestwind @seanpreston @pattisdr one concern I have about including pagination when a user searches on the front end using that "typeahead" method we discussed. If this UI is paginated, would a search only return matching connections that are on that page? If so, that would be considered an incomplete search by the end user. May not be the case but want to bring it up just in case. 

Also, may be good to run this by the design team quickly because this would be a change with their designs they provided us.

Reply:
@adriaaaa , if we had pagination here, you're correct in that if the amount of data exceeds a certain threshold, a search would only return 1 page of connections, with the need for another api call to return the next page. As we scale and/or add more data to this endpoint, @seanpreston is right that pagination will probably be necessary in the future.

Reply:
>  would a search only return matching connections that are on that page?

Search via the API is done at the database level, which always has full context of the available dataset, so if the search term is submitted to the API then what you're worried about here won't be a problem.

If the search isn't submitted via the API, and instead done via the client-side code checking results from the API, then including pagination means we have to be sure that the client has a full collection of the results. We can increase the page size of what's requested in this http request to guarantee we'll fetch the whole lot in the first go, so this shouldn't be an issue.